modules/dcache/src/main/java/org/dcache/pool/statistics/IoStatisticsChannel.java

```diff
@@ -0,0 +1,168 @@
+package org.dcache.pool.statistics;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.dcache.pool.repository.RepositoryChannel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.SyncFailedException;
+import java.nio.ByteBuffer;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.channels.SeekableByteChannel;
+import java.nio.channels.WritableByteChannel;
+
+/**
+ * This class decorates any RepositoryChannel and adds the function of collecting data for the IoStatistics
+ */
+public class IoStatisticsChannel implements RepositoryChannel {
+
+    private static final Logger LOGGER =
+            LoggerFactory.getLogger(IoStatisticsChannel.class);
+
+    /**
+     * Inner channel to which all operations are delegated.
+     */
+    @VisibleForTesting
```

Comment:
I don't believe this is going to work.  You need to make the field package visible and put the test class in the same package.  This annotation is for informational purposes only.